src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1326,9 +1327,32 @@ private static string GetParamDefaultValueAsString(object? value, string objectT
             {
                 case null:
                     return $"default({objectTypeAsStr})";
-                case bool boolVal:
-                    return ToCSharpKeyword(boolVal);
+                case bool @bool:
+                    return ToCSharpKeyword(@bool);
+                case string @string:
+                    return @$"""{@string}""";
+                case char @char:
+                    return @$"'{@char}'";
+                case double.NegativeInfinity:
+                    return "double.NegativeInfinity";
+                case double.PositiveInfinity:
+                    return "double.PositiveInfinity";
+                case double.NaN:
+                    return "double.NaN";
+                case float.NegativeInfinity:
+                    return "float.NegativeInfinity";
+                case float.PositiveInfinity:
+                    return "float.PositiveInfinity";
+                case float.NaN:
+                    return "float.NaN";
+                case double @double:
+                    return @double.ToString(JsonConstants.DoubleFormatString, CultureInfo.InvariantCulture);
+                case float @float:
+                    return @float.ToString(JsonConstants.SingleFormatString, CultureInfo.InvariantCulture);
+                case decimal @decimal:
+                    return @decimal.ToString(CultureInfo.InvariantCulture);
                 default:
+                    // Assume this is a number.
```

Comment:
Is this approach of checking the various special cases the same as the reflection-based code? I assume the reflection-based code doesn't need to generate a string and can just use the raw ParameterValue?

Reply:
Yeah the reflection-based code only has to cast the boxed default value to the right target type (and doesn't need to write valid C#). It does this using generics in manner that doesn't need special-casing.